### PR TITLE
[improvement](compaction) delete num based compaction policy

### PR DESCRIPTION
**Pull request by yixiutt**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by Gabriel39 — DISMISSED** [review]
LGTM

File: be/src/olap/cumulative_compaction_policy.h [hunk 1]
```diff
@@ -248,13 +193,7 @@ class CumulativeCompactionPolicyFactory {
...
     /// Static factory function. It can product different policy according to the `policy` parameter and use tablet ptr
     /// to construct the policy. Now it can product size based and num based policies.
-    static std::shared_ptr<CumulativeCompactionPolicy> create_cumulative_compaction_policy(
-            std::string policy);
-
-private:
-    /// It is a static function to help to check the policy config and convert to CompactionPolicy enum variable
-    static void _parse_cumulative_compaction_policy(std::string policy,
-                                                    CompactionPolicy* policy_type);
+    static std::shared_ptr<CumulativeCompactionPolicy> create_cumulative_compaction_policy();
```
Comment (zhannngchen) [inline be/src/olap/cumulative_compaction_policy.h:196]: The comment of this method should also be updated.

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**